src/sql/connection/connection.py

```diff
@@ -1187,4 +1174,27 @@ def set_sqlalchemy_isolation_level(conn):
         return False
 
 
+def detect_duckdb_summarize_or_select(query):
+    """
+    Checks if the SQL query is a DuckDB SELECT or SUMMARIZE statement.
+
+    Note:
+    Assumes there is only one SQL statement in the query.
+    """
+    statements = sqlparse.parse(query)
+    if statements:
+        assert len(statements) == 1
+        stype = statements[0].get_type()
+        if stype == "SELECT":
+            return True
+        elif stype == "UNKNOWN":
+            # Further analysis is required
+            sql_stripped = sqlparse.format(query, strip_comments=True)
+            words = sql_stripped.split()
+            return len(words) > 0 and (
```

Comment:
can you add a test with an empty SQL query? (would `words[0]` break?)

Reply:
I have a test case with an empty query https://github.com/ploomber/jupysql/pull/895/files#diff-ecfdf2a1f3adc925d3c32b8aa52540474a10a4160d552f281d7dbd85a0260170R1200

but do you mean that the `len(words) > 0` check is not required?

Reply:
I mean passing `""` (empty string) as input, would that break the test?

Reply:
The location of it moved slightly but here is the empty string test: https://github.com/ploomber/jupysql/pull/895/files#diff-ecfdf2a1f3adc925d3c32b8aa52540474a10a4160d552f281d7dbd85a0260170R1205

In the repl:
```
In [1]: from sql.connection import detect_duckdb_summarize_or_select

In [2]: detect_duckdb_summarize_or_select("")
Out[2]: False
```

It's because `sqlparse.parse("")` returns an empty collection
```
In [3]: import sqlparse

In [4]: sqlparse.parse("")
Out[4]: ()
```